jetty-server/src/test/java/org/eclipse/jetty/server/EchoHandler.java

```diff
@@ -0,0 +1,91 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.util.IteratingCallback;
+import org.eclipse.jetty.util.StringUtil;
+
+/**
+ * Dump request handler.
+ * Dumps GET and POST requests.
+ * Useful for testing and debugging.
+ */
+public class EchoHandler extends Handler.Abstract
```

Comment:
@sbordet @lorban @lachlan-roberts  This is a good place to start looking jetty-12 Handlers.   This is a fully async handler that echos the content-type, content-length, the content itself and any trailers received.